### PR TITLE
Replace ui.language.isRtl Redux state with HOC that depends on i18n-calypso

### PR DESCRIPTION
**Pull request by jsnajdr**
Several basic Calypso components like `Popover` depend on the `isRtl` Redux selector from `state/selectors/is-rtl` to render correctly in an RTL environment. Typically, tooltip positioning, inline CSS styles and arrow icons depend on the LTR/RTL flag.

Depending on Redux makes these components harder to extract to libraries and reuse. This PR removes the `isRtl` selector and the associated Redux state in favor of a `components/rtl` helper module that observes the state of the `i18n-calypso` object directly. There is a `useRtl` hook and a `withRtl` HOC.

**How to test:**
Enable RTL language like Hebrew and check that the `Chart` in Stats has all RTL features working:
- the X axis with labels is correctly rendered left-to-right
- the tooltip alignment is flipped when too near to the chart edge

Screenshot 2019-10-01 at 12 00 38

### CODE REVIEW

**Comment by jsnajdr** [comment]
ICFY reports some very nice bundle size reductions: I honestly don't even know where they come from 🙂 Removing a reducer should affect the entrypoint size, but not the sections and async-loaded components?!

**Review by sgomes — APPROVED** [review]
Thank you for this PR, @jsnajdr! You've identified a really good opportunity here to extract redundant state from Redux 👍 

Testing also appears to indicate that this is working correctly.

I added some minor comments, but this is good to go, as far as I'm concerned 🎉

File: client/components/chart/bar.jsx [hunk 1]
```diff
@@ -37,18 +36,13 @@ export default class ChartBar extends React.PureComponent {
...
+		const shouldFlip = barOffset > chartWidth - 230 && barOffset + barWidth > 230;
 
-		let tooltipPosition = isRtl ? 'bottom left' : 'bottom right';
-
-		if ( barOffset + 230 > chartWidth && barOffset + barWidth - 230 > 0 ) {
-			tooltipPosition = isRtl ? 'bottom right' : 'bottom left';
-		}
-
-		return tooltipPosition;
+		return shouldFlip ? 'bottom left' : 'bottom right';
```
[the remark this answers is not in the document]
Comment (sgomes) [reply client/components/chart/bar.jsx]: Confirmed the tooltip still opens in the correct direction (to the left in RTL).

File: client/components/rtl/index.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	const [ isRtl, setRtl ] = useState( isCurrentLanguageRtl );
+
+	useEffect( () => {
+		const onChange = () => setRtl( isCurrentLanguageRtl );
+		i18n.on( 'change', onChange );
+		return () => i18n.off( 'change', onChange );
+	} );
+
+	return isRtl;
+}
```
[the remark this answers is not in the document]
Comment (jsnajdr) [reply client/components/rtl/index.js]: That's a good point! I'll try to use the [useSubscription](https://www.npmjs.com/package/use-subscription) package here (it was published by React team not so long time ago)

[the remark this answers is not in the document]
Comment (jsnajdr) [reply client/components/rtl/index.js]: @sgomes I pushed a commit (9c9f60b) with a rewrite of the `useRtl` hook to use the `use-subscription` package (recently published by the React team)

**Comment by sgomes** [comment]
LGTM, assuming they implemented `useSubscription` the right way! 😄

File: client/components/rtl/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (sgomes) [reply client/components/rtl/index.js]: LGTM, assuming they implemented useSubscription the right way! 😄

[the remark this answers is not in the document]
Comment (jsnajdr) [reply client/components/rtl/index.js]: We gave them a lot of good advice, didn't we? 😆

**Pull request merged**